   # Dummy file to track dependency.
--- third_party/WebKit/Source/platform/v8_inspector/v8_inspector.gyp.orig	2016-06-03 12:31:49.844954196 +0000
+++ third_party/WebKit/Source/platform/v8_inspector/v8_inspector.gyp	2016-06-03 12:32:01.869198425 +0000
@@ -60,7 +60,6 @@
         {
           'action_name': 'generateV8InspectorProtocolBackendSources',
           'inputs': [
-            '<@(jinja_module_files)',
             # The python script in action below.
             '../inspector_protocol/CodeGenerator.py',
             # Input files for the script.
