--- a/src/main.c
+++ b/src/main.c
@@ -1615,7 +1615,7 @@
                        "  h2o [options]\n"
                        "\n"
                        "Options:\n"
-                       "  -c, --conf FILE    configuration file (default: h2o.conf)\n"
+                       "  -c, --conf FILE    configuration file (default: /etc/h2o/h2o.conf)\n"
                        "  -m, --mode <mode>  specifies one of the following mode\n"
                        "                     - worker: invoked process handles incoming connections\n"
                        "                               (default)\n"
@@ -1624,7 +1624,7 @@
                        "                               the errors are logged to the file instead of\n"
                        "                               being emitted to STDERR\n"
                        "                     - master: invoked process becomes a master process (using\n"
-                       "                               the `share/h2o/start_server` command) and spawns\n"
+                       "                               the `/usr/share/h2o/start_server` command) and spawns\n"
                        "                               a worker process for handling incoming\n"
                        "                               connections. Users may send SIGHUP to the master\n"
                        "                               process to reconfigure or upgrade the server.\n"
@@ -1633,8 +1633,8 @@
                        "  -v, --version      prints the version number\n"
                        "  -h, --help         print this help\n"
                        "\n"
-                       "Please refer to the documentation under `share/doc/h2o` (or available online at\n"
-                       "http://h2o.examp1e.net/) for how to configure the server.\n"
+                       "Please refer to the documentation under `/usr/share/doc/h2o` (or available online at\n"
+                       "https://h2o.examp1e.net/) for how to configure the server.\n"
                        "\n");
                 exit(0);
                 break;
