--- nsd-control.c.orig	2015-10-29 17:31:41.000000000 +0900
+++ nsd-control.c	2015-12-31 10:37:54.454981962 +0900
@@ -125,12 +125,16 @@
         ctx = SSL_CTX_new(SSLv23_client_method());
 	if(!ctx)
 		ssl_err("could not allocate SSL_CTX pointer");
+#ifndef OPENSSL_NO_SSL2
         if((SSL_CTX_set_options(ctx, SSL_OP_NO_SSLv2) & SSL_OP_NO_SSLv2)
 		!= SSL_OP_NO_SSLv2)
 		ssl_err("could not set SSL_OP_NO_SSLv2");
+#endif
+#ifndef OPENSSL_NO_SSL3
         if((SSL_CTX_set_options(ctx, SSL_OP_NO_SSLv3) & SSL_OP_NO_SSLv3)
 		!= SSL_OP_NO_SSLv3)
 		ssl_err("could not set SSL_OP_NO_SSLv3");
+#endif
 	if(!SSL_CTX_use_certificate_file(ctx,c_cert,SSL_FILETYPE_PEM) ||
 		!SSL_CTX_use_PrivateKey_file(ctx,c_key,SSL_FILETYPE_PEM)
 		|| !SSL_CTX_check_private_key(ctx))
--- remote.c.orig	2015-10-29 17:31:41.000000000 +0900
+++ remote.c	2015-12-31 10:40:40.408051933 +0900
@@ -262,6 +262,7 @@
 		free(rc);
 		return NULL;
 	}
+#ifndef OPENSSL_NO_SSL2
 	/* no SSLv2, SSLv3 because has defects */
 	if((SSL_CTX_set_options(rc->ctx, SSL_OP_NO_SSLv2) & SSL_OP_NO_SSLv2)
 		!= SSL_OP_NO_SSLv2){
@@ -269,12 +270,15 @@
 		daemon_remote_delete(rc);
 		return NULL;
 	}
+#endif
+#ifndef OPENSSL_NO_SSL3
 	if((SSL_CTX_set_options(rc->ctx, SSL_OP_NO_SSLv3) & SSL_OP_NO_SSLv3)
 		!= SSL_OP_NO_SSLv3){
 		log_crypto_err("could not set SSL_OP_NO_SSLv3");
 		daemon_remote_delete(rc);
 		return NULL;
 	}
+#endif
 	s_cert = cfg->server_cert_file;
 	s_key = cfg->server_key_file;
 	VERBOSITY(2, (LOG_INFO, "setup SSL certificates"));
